<testsuite name="Mocha Tests" tests="18" failures="1" errors="1" skipped="0" timestamp="Tue, 16 Aug 2016 08:57:40 GMT" time="4.608">
<testcase classname="end to end on page load" name="displays TODO title" time="0.362"/>
<testcase classname="end to end on page load" name="displays empty TODO list" time="0.107"/>
<testcase classname="end to end on page load" name="displays an error if the request fails" time="0.113"/>
<testcase classname="end to end on create todo item" name="clears the input field" time="0.336"/>
<testcase classname="end to end on create todo item" name="adds the todo item to the list" time="0.23"/>
<testcase classname="end to end on create todo item" name="displays an error if the request fails" time="0.191"/>
<testcase classname="end to end on create todo item" name="can be done multiple times" time="0.293"/>
<testcase classname="end to end on delete todo item" name="deletes the todo item to the list" time="0.268"/>
<testcase classname="end to end on update todo item" name="updates the todo item to the list" time="0.182"><failure>driver.findElements(...).click is not a function
TypeError: driver.findElements(...).click is not a function
    at Object.module.exports.updateTodo (C:\dev\todo\test\e2eHelpers.js:95:50)
    at C:\dev\todo\test\e2e.js:87:36
    at ManagedPromise.invokeCallback_ (C:\dev\todo\node_modules\selenium-webdriver\lib\promise.js:1379:14)
    at TaskQueue.execute_ (C:\dev\todo\node_modules\selenium-webdriver\lib\promise.js:2913:14)
    at TaskQueue.executeNext_ (C:\dev\todo\node_modules\selenium-webdriver\lib\promise.js:2896:21)
    at C:\dev\todo\node_modules\selenium-webdriver\lib\promise.js:2775:27
    at C:\dev\todo\node_modules\selenium-webdriver\lib\promise.js:639:7
From: Task: end to end on update todo item updates the todo item to the list
    at Context.ret (C:\dev\todo\node_modules\selenium-webdriver\testing\index.js:128:10)
    at callFnAsync (C:\dev\todo\node_modules\mocha\lib\runnable.js:349:8)
    at Test.Runnable.run (C:\dev\todo\node_modules\mocha\lib\runnable.js:301:7)
    at Runner.runTest (C:\dev\todo\node_modules\mocha\lib\runner.js:422:10)
    at C:\dev\todo\node_modules\mocha\lib\runner.js:528:12
    at next (C:\dev\todo\node_modules\mocha\lib\runner.js:342:14)
    at C:\dev\todo\node_modules\mocha\lib\runner.js:352:7
    at next (C:\dev\todo\node_modules\mocha\lib\runner.js:284:14)
    at Immediate._onImmediate (C:\dev\todo\node_modules\mocha\lib\runner.js:320:5)</failure></testcase>
<testcase classname="server get list of todos" name="responds with status code 200" time="0.014"/>
<testcase classname="server get list of todos" name="responds with a body encoded as JSON in UTF-8" time="0.01"/>
<testcase classname="server get list of todos" name="responds with a body that is a JSON empty array" time="0.004"/>
<testcase classname="server create a new todo" name="responds with status code 201" time="0.006"/>
<testcase classname="server create a new todo" name="responds with the location of the newly added resource" time="0.004"/>
<testcase classname="server create a new todo" name="inserts the todo at the end of the list of todos" time="0.019"/>
<testcase classname="server delete a todo" name="responds with status code 404 if there is no such item" time="0.005"/>
<testcase classname="server delete a todo" name="responds with status code 200" time="0.006"/>
<testcase classname="server delete a todo" name="removes the item from the list of todos" time="0.014"/>
</testsuite>
